packages/integration-tests/src/amplify_auth_credentials_factory.ts

```diff
@@ -35,72 +44,85 @@ export class AmplifyAuthCredentialsFactory {
   }
 
   getNewAuthenticatedUserCredentials = async (): Promise<IamCredentials> => {
-    const username = `amplify-backend-${shortUuid()}@amazon.com`;
-    const temporaryPassword = `Test1@Temp${shortUuid()}`;
-    const password = `Test1@${shortUuid()}`;
-    await this.cognitoIdentityProviderClient.send(
-      new AdminCreateUserCommand({
-        Username: username,
-        TemporaryPassword: temporaryPassword,
-        UserPoolId: this.userPoolId,
-        MessageAction: 'SUPPRESS',
-      })
-    );
-
-    Amplify.configure({
-      Auth: {
-        Cognito: {
-          userPoolId: this.userPoolId,
-          userPoolClientId: this.userPoolClientId,
-          identityPoolId: this.identityPoolId,
-          allowGuestAccess: this.allowGuestAccess,
+    await this.lock.acquire();
+    try {
+      const username = `amplify-backend-${shortUuid()}@amazon.com`;
+      const temporaryPassword = `Test1@Temp${shortUuid()}`;
+      const password = `Test1@${shortUuid()}`;
+      await this.cognitoIdentityProviderClient.send(
+        new AdminCreateUserCommand({
+          Username: username,
+          TemporaryPassword: temporaryPassword,
+          UserPoolId: this.userPoolId,
+          MessageAction: 'SUPPRESS',
+        })
+      );
+
+      Amplify.configure({
+        Auth: {
+          Cognito: {
+            userPoolId: this.userPoolId,
+            userPoolClientId: this.userPoolClientId,
+            identityPoolId: this.identityPoolId,
+            allowGuestAccess: this.allowGuestAccess,
+          },
         },
-      },
-    });
+      });
 
-    const signInResult = await auth.signIn({
-      username,
-      password: temporaryPassword,
-    });
+      // in case there's already signed user in the session.
+      await auth.signOut();
```

Comment:
will this throw if a user is not logged in?

Reply:
If checks pass on this PR then yes.
I'll wait for confirmation before merging. Or add some defensive logic otherwise.